src/foam/u2/wizard/StepWizardController.js

```diff
@@ -0,0 +1,321 @@
+/**
+ * @license
+ * Copyright 2021 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.u2.wizard',
+  name: 'StepWizardController',
+
+  requires: [
+    'foam.core.FObject',
+    'foam.u2.wizard.WizardPosition',
+    'foam.u2.wizard.StepWizardConfig'
+  ],
+
+  properties: [
+    {
+      class: 'String',
+      name: 'title'
+    },
+    {
+      name: 'config',
+      class: 'FObjectProperty',
+      of: 'foam.u2.wizard.StepWizardConfig',
+      factory: function() {
+        return this.StepWizardConfig.create();
+      }
+    },
+    {
+      name: 'wsub',
+      class: 'FObjectProperty',
+      of: 'FObject',
+      description: `
+        Subscription for listeners of wizardlets' state. This is replaced if the
+        list of wizardlets is updated.
+      `
+    },
+    {
+      name: 'wizardlets',
+      class: 'FObjectArray',
+      of: 'foam.u2.wizard.Wizardlet',
+      postSet: function (_, n) {
+        this.setupWizardletListeners(n);
+      }
+    },
+    {
+      name: 'wizardPosition',
+      documentation: `
+        Wizardlet position
+      `,
+      class: 'FObjectProperty',
+      of: 'foam.u2.wizard.WizardPosition',
+      factory: function() {
+        return this.WizardPosition.create({
+          wizardletIndex: 0,
+          sectionIndex: 0,
+        });
+      }
+    },
+
+    {
+      name: 'nextScreen',
+      expression: function(wizardPosition) {
+        return this.nextAvailable(wizardPosition, this.positionAfter.bind(this));
+      }
+    },
+    {
+      name: 'previousScreen',
+      expression: function(wizardPosition) {
+        return this.nextAvailable(wizardPosition, this.positionBefore.bind(this));
+      }
+    },
+    {
+      name: 'visitedWizardlets',
+      class: 'FObjectArray',
+      of: 'foam.u2.wizard.Wizardlet'
+    },
+
+    // Convenience properties
+    {
+      name: 'currentWizardlet',
+      expression: function(wizardlets, wizardPosition) {
+        var current = wizardlets[wizardPosition.wizardletIndex];
```

Comment:
If visitedWizardlets was an int array you can just compare ints instead of objects here.

Reply:
Or you can create a boolean array with length of wizardlets and set visited wizardlets as true and therefore no comparison here.

Reply:
object comparisons in javascript are reference comparisons - they perform just as well as the int or bool comparisons you suggest